src/services/extension-registry-provider.js

```diff
@@ -19,7 +21,41 @@
             reduce = utils.reduce,
             ownKeys = utils.ownKeys,
             toArray = utils.toArray,
-            isFunction = utils.isFunction;
+            isFunction = utils.isFunction,
+            isNaN = utils.isNaN,
+            isString = utils.isString,
+            isNumber = utils.isNumber,
+            isUndefined = utils.isUndefined,
+            // TODO: externalize this validator
+            // so that a custom sorter could be provided
+            invalidVal = -9999,
+            strToUsableNum = function(val) {
+              var test = Number(val);
+              return isNaN(test) ?
+                      invalidVal :
+                      test;
+            },
+            // ensures any provided weight is usable.
+            // see the unit test for weight
+            // - numbers returned as numbers
+            // - undefined returned as -1 (explicit 0 has greater weight)
+            // - strings returned as numbers || -9999 (invalid)
+            // - all other objects returned as -9999 (invalid)
+            // this function is intentionally resilient as it is assumed
+            // extensions could be registered by various parties.  Those
+            // who use sensible weights should be able to trust sorting
+            // placement
+            toWeight = function(value) {
+              return isNaN(value) ?
+                      invalidVal :
+                      isNumber(value) ?
+                        value :
+                        isString(value) ?
+                          strToUsableNum(value) :
+                          isUndefined(value) ?
+                            -1 :
+                            invalidVal;
```

Comment:
Nested ternary checks are really hard to read and understand.

I don't see a reason to explicitly check for every type. Make the default weight 1, then

```js
toWeight = function(value) {
  if (value === null || value === undefined) {
    return 1;
  }

  value = praseInt(value, 10);
  if (isNaN(value)) {
   return 0;
  }

  return value;
};
```

I'm not sure it's even worth handling invalid values differently than default values, though. It could be as simple as

```js
toWeight = function(value) {
  value = praseInt(value, 10);
  if (isNaN(value)) {
   return 0;
  }

  return value;
};
```